src/task-actions.ts

```diff
@@ -156,10 +156,14 @@ export async function starknetDeployAction(args: any, hre: HardhatRuntimeEnviron
     let statusCode = 0;
     const txHashes: string[] = [];
     for (let artifactsPath of artifactsPaths) {
-        if (!path.isAbsolute(artifactsPath)) {
+
+        // Check if input is the name of the contract and not a path
+        if (artifactsPath === path.basename(artifactsPath)) {
```

Comment:
What does this introduce? When deploying, contract name (with extension) can be provided? Is that the only case supported with the addition of these lines? And is this tested?